rsocket/internal/SetupResumeAcceptor.cpp

```diff
@@ -70,120 +71,94 @@ SetupResumeAcceptor::~SetupResumeAcceptor() {
+        close(std::move(transport), std::move(exn));
+        break;
       }
 
-      ResumeParameters resumeParams(
-          std::move(resumeFrame.token_),
-          resumeFrame.lastReceivedServerPosition_,
-          resumeFrame.clientPosition_,
-          ProtocolVersion(
-              resumeFrame.versionMajor_, resumeFrame.versionMinor_));
-
-      if (frameSerializer->protocolVersion() != resumeParams.protocolVersion) {
-        transport->outputFrameOrEnqueue(frameSerializer->serializeOut(
-            Frame_ERROR::badSetupFrame("invalid protocol version")));
-        closeAndRemoveConnection(
-            std::move(transport),
-            std::runtime_error("invalid protocol version"));
+      ResumeParameters params(
+          std::move(frame.token_),
+          frame.lastReceivedServerPosition_,
+          frame.clientPosition_,
+          ProtocolVersion(frame.versionMajor_, frame.versionMinor_));
+
+      if (serializer->protocolVersion() != params.protocolVersion) {
+        std::runtime_error exn{"RESUME frame has invalid protocol version"};
+        transport->outputFrameOrEnqueue(
+            serializer->serializeOut(Frame_ERROR::badSetupFrame(exn.what())));
+        close(std::move(transport), std::move(exn));
         break;
       }
 
-      if (!onResume(transport, std::move(resumeParams))) {
-        transport->outputFrameOrEnqueue(frameSerializer->serializeOut(
-            Frame_ERROR::rejectedResume("Resumption Failure")));
-        // We are currently in FrameTransport's callstack where it would
-        // have already acquired a recursive_mutex.  After the following block
-        // of code, FrameTransport goes out of scope and will get destroyed. But
-        // we cant destroy it while holding a recursive_mutex.  So schedule this
-        // block of code to be executed after callstack has unwinded.
-        eventBase_->runInEventBaseThread(
-            [ this, transport = std::move(transport) ] {
-              closeAndRemoveConnection(
-                  std::move(transport),
-                  std::runtime_error("Resumption Failure"));
-            });
-      } else {
+      if (onResume(transport, std::move(params))) {
         connections_.erase(transport);
+        return;
       }
 
+      std::runtime_error exn{"Resumption failure"};
+
+      transport->outputFrameOrEnqueue(
+          serializer->serializeOut(Frame_ERROR::rejectedResume(exn.what())));
+
+      close(std::move(transport), std::move(exn));
       break;
     }
 
-    case FrameType::CANCEL:
-    case FrameType::ERROR:
-    case FrameType::KEEPALIVE:
-    case FrameType::LEASE:
-    case FrameType::METADATA_PUSH:
-    case FrameType::REQUEST_CHANNEL:
-    case FrameType::REQUEST_FNF:
-    case FrameType::REQUEST_N:
-    case FrameType::REQUEST_RESPONSE:
-    case FrameType::REQUEST_STREAM:
-    case FrameType::RESERVED:
-    case FrameType::PAYLOAD:
-    case FrameType::RESUME_OK:
-    case FrameType::EXT:
```

Comment:
They shouldn't be necessary, as there's a `default` label that captures all of them.  I don't remember a warning included in `-Wextra` that would trigger that, are you thinking of `-Weverything` instead?

Also note that there are no warnings in the builds despite us using `-Wall -Wextra -Werror` for code in `rsocket/`.